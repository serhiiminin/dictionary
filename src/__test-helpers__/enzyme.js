import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';

const shallowWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()
  return shallow(tree, { context })
}

const mountWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()

  return mount(tree, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  })
}

export { shallowWithTheme, mountWithTheme }
