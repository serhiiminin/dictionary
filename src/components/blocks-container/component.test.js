import React from 'react';
import { shallow } from 'enzyme';
import BlocksContainer from './component';

describe('Blocks Container', () => {
  test('render', () => {
    const wrapper = shallow(
      <BlocksContainer>
        Anything
      </BlocksContainer>
    );

    expect(wrapper.contains('Anything')).toMatchSnapshot();
  });
});
