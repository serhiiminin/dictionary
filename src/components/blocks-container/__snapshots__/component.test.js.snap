// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blocks Container render 1`] = `
<div
  className={undefined}
>
  Anything
</div>
`;
